vpc/service/allocate.go

```diff
@@ -23,67 +22,98 @@ import (
-		})
-		if err != nil {
-			return nil, ec2wrapper.HandleEC2Error(err, span)
-		}
-		if l := len(describeNetworkInterfacesOutput.NetworkInterfaces); l != 1 {
-			err = fmt.Errorf("Unexpected number of interfaces returned from DescribeNetworkInterfacesWithContext: %d", l)
-			span.SetStatus(traceStatusFromError(err))
-			return nil, err
-		}
-		return describeNetworkInterfacesOutput.NetworkInterfaces[0], nil
+func allocateStaticSubnetCidrReservations(ctx context.Context, tx *sql.Tx, session *ec2wrapper.EC2Session, v4addr net.IP, v6addr net.IP, alloc allocation) error {
+	v6output, err := session.CreateSubnetCidrReservation(ctx, ec2.CreateSubnetCidrReservationInput{
+		Cidr:            aws.String(fmt.Sprintf("%s/128", v6addr.String())),
+		Description:     aws.String(staticReservationDescription),
+		ReservationType: aws.String("explicit"),
+		SubnetId:        aws.String(alloc.subnet),
+	})
+	if err != nil {
+		return fmt.Errorf("Cannot make reservation for IPv6 addr: %w", err)
 	}
 
-	createNetworkInterfaceOutput, err := ec2client.CreateNetworkInterfaceWithContext(ctx, &ec2.CreateNetworkInterfaceInput{
-		Description:                    aws.String(staticDummyInterfaceDescription),
-		SubnetId:                       aws.String(subnetID),
-		Ipv6AddressCount:               aws.Int64(maxAddressesPerInterface),
-		SecondaryPrivateIpAddressCount: aws.Int64(maxAddressesPerInterface - 1),
+	v4output, err := session.CreateSubnetCidrReservation(ctx, ec2.CreateSubnetCidrReservationInput{
+		Cidr:            aws.String(fmt.Sprintf("%s/32", v4addr.String())),
+		Description:     aws.String(staticReservationDescription),
+		ReservationType: aws.String("explicit"),
+		SubnetId:        aws.String(alloc.subnet),
 	})
-	if err != nil && err != sql.ErrNoRows {
-		return nil, ec2wrapper.HandleEC2Error(err, span)
+	if err != nil {
+		return fmt.Errorf("Cannot make reservation for IPv4 addr: %w", err)
 	}
 
-	eniID = aws.StringValue(createNetworkInterfaceOutput.NetworkInterface.NetworkInterfaceId)
-	_, err = tx.ExecContext(ctx, "INSERT INTO static_enis(eni_id, subnet_id) VALUES ($1, $2)", eniID, subnetID)
+	_, err = tx.ExecContext(ctx,
+		"INSERT INTO subnet_cidr_reservations_v6(reservation_id, subnet_id, prefix, type, description) VALUES ($1, $2, $3, $4, $5) ON CONFLICT DO NOTHING ",
+		aws.StringValue(v6output.SubnetCidrReservation.SubnetCidrReservationId),
+		alloc.subnetID,
+		aws.StringValue(v6output.SubnetCidrReservation.Cidr),
+		aws.StringValue(v6output.SubnetCidrReservation.ReservationType),
+		aws.StringValue(v6output.SubnetCidrReservation.Description),
+	)
 	if err != nil {
-		err = errors.Wrap(err, "Unable to save newly statically created ENI")
-		span.SetStatus(traceStatusFromError(err))
-		return nil, err
+		return fmt.Errorf("Could not insert subnet CIDR v6 reservation %s: %w", aws.StringValue(v6output.SubnetCidrReservation.SubnetCidrReservationId), err)
 	}
 
-	return createNetworkInterfaceOutput.NetworkInterface, nil
+	_, err = tx.ExecContext(ctx,
```

Comment:
I would say a TODO here (and you don't need to do it as part of this PR) is error recovery semantics.